### PR TITLE
Switch from 'failure' to 'anyhow'

### PR DESCRIPTION
**Pull request by Allen-Webb**
This switchies from the 'failure' crate to the 'anyhow' crate to resolve issue #40.

It also fixes deprecated useage of protobuf::parse_from_bytes to Message::parse_from_bytes.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "protoc-grpcio"
-version = "3.0.0"
+version = "4.0.0"
```
Comment (Allen-Webb) [inline Cargo.toml:3]: This may not be necessary, but I bumped the version to be safe since `pub type CompileError = ::failure::Error;` leaks a type that changes before and after this change resulting in a theoretical API breakage, but in practice it shouldn't require much if any refactoring.

File: src/lib.rs [hunk 2]
```diff
@@ -253,7 +253,7 @@ where
         .context("failed to read descriptor set")?;
 
     let descriptor_set =
-        protobuf::parse_from_bytes::<descriptor::FileDescriptorSet>(&serialized_descriptor_set)
+        descriptor::FileDescriptorSet::parse_from_bytes(&serialized_descriptor_set)
```
Comment (Allen-Webb) [inline src/lib.rs:256]: It might make sense to split this into a different commit, but I included it because I have compilation set up to fail when deprecated functions are used.

File: Cargo.toml (hunk 1 above)
Comment (mtp401) [reply Cargo.toml:3]: I think the major bump here is needed for the reasons you pointed surrounding API incompatibility, so, this is the correct move IMO.

**Review by mtp401 — APPROVED** [review]
Thanks for the PR!
